app/models/fe/question_set.rb

```diff
@@ -106,7 +106,9 @@ def posted_values(param)
         end
       elsif param.kind_of?(Hash)
         # from Hash with multiple answers per question
-        values = param.values.map {|v| CGI.unescape(v)}
+        # If value is also a hash, use the value hash without escaping or anything,
+        # so that custom elements can be implemented by handling set_response.
+        values = param.values.map {|v| v.is_a?(Hash) ? v : CGI.unescape(v)}
```

Comment:
Is this a security concern? Map one more ply and unescape the values of each v.values, perhaps?

Also, I notice line 103 doesn't return an array, but that's not in this patch.